core/src/main/java/org/apache/gravitino/cache/CaffeineEntityCache.java

```diff
@@ -228,7 +241,8 @@ public <E extends Entity & HasIdentifier> void put(E entity) {
         () -> {
           invalidateOnKeyChange(entity);
           NameIdentifier identifier = getIdentFromEntity(entity);
-          EntityCacheKey entityCacheKey = EntityCacheKey.of(identifier, entity.type());
+          EntityCacheRelationKey entityCacheKey =
+              EntityCacheRelationKey.of(identifier, entity.type());
```

Comment:
How do you know that for some entity types, we need to update the entity when the related entity is updated/deleted? For example, if the role is deleted, we need to update the user entity, here I don't see the specific logic to handle this. I think we should have a specific logic to check the entity type and do some specific logic. It is hard to make it generic.